app/models/task.rb

```diff
@@ -41,13 +42,19 @@ def accept_notes
     def update_job
       @job = self.job
       @job.update(done: @job.tasks.select{|x| x.status != "accepted"}.empty?)
-      if @job.done? && @job.assignment.awards.count < 3 && @job.assignment.assignment_type != "test"
-        @job.awards.create(rank: @job.assignment.awards.count + 1)
-
+      if @job.done?
+        if @job.assignment.awards.count < 3 && @job.assignment.assignment_type != "test"
+          @job.awards.create(rank: @job.assignment.awards.count + 1)
+        end
         assignment = @job.assignment
         if assignment.jobs.where(done: false).empty?
           assignment.update(done: true)
         end
+      else
+        @job.assignment.update(done: false)
+        if @job.awards.count > 0
+          @job.awards.first.destroy
```

Comment:
There are two opportunities: 
1) Teacher adds new tasks to homeworks, so ruby updates every job of this homework. If job had award before update, award would be destroyed. And this would be done for every job.
2) Teacher accepted last task of some student's job, but after some time understood that it was a mistake. So he cancelled this task. If the job had an award, award would be destroyed.